ogre2/src/Ogre2RayQuery.cc

```diff
@@ -126,17 +131,75 @@ void Ogre2RayQuery::SetFromCamera(const CameraPtr &_camera,
 }
 
 //////////////////////////////////////////////////
-RayQueryResult Ogre2RayQuery::ClosestPoint(bool _forceSceneUpdate)
+void Ogre2RayQuery::SetFromCamera(const WideAngleCameraPtr &_camera,
+                                  uint32_t _faceIdx,
+                                  const math::Vector2d &_coord)
 {
-  RayQueryResult result;
+  // convert to nomalized screen pos for ogre
+  math::Vector2d screenPos((_coord.X() + 1.0) / 2.0, (_coord.Y() - 1.0) / -2.0);
+
+  Ogre2WideAngleCameraPtr camera =
+    std::dynamic_pointer_cast<Ogre2WideAngleCamera>(_camera);
+  this->dataPtr->camera.reset();
+
+  Ogre::Ray ray = camera->CameraToViewportRay(screenPos, _faceIdx);
+
+  auto originMath = Ogre2Conversions::Convert(ray.getOrigin());
+  if (originMath.IsFinite())
+  {
+    this->origin = originMath;
+  }
+  else
+  {
+    gzwarn << "Attempted to set non-finite origin from camera ["
+           << camera->Name() << "]" << std::endl;
+  }
+
+  auto directionMath = Ogre2Conversions::Convert(ray.getDirection());
+  if (directionMath.IsFinite())
+  {
+    this->direction = directionMath;
+  }
+  else
+  {
+    gzwarn << "Attempted to set non-finite direction from camera ["
+           << camera->Name() << "]" << std::endl;
+  }
 
+  this->dataPtr->imgPos.X() =
+    static_cast<int>(screenPos.X() * _camera->ImageWidth());
+  this->dataPtr->imgPos.Y() =
+    static_cast<int>(screenPos.Y() * _camera->ImageHeight());
+}
 
+//////////////////////////////////////////////////
+void Ogre2RayQuery::SetPreferGpu(bool _preferGpu)
+{
+  this->dataPtr->preferGpu = _preferGpu;
+}
+
+//////////////////////////////////////////////////
+bool Ogre2RayQuery::UsesGpu() const
+{
 #ifdef __APPLE__
```

Comment:
I think GPU on macOS should now work here as well right? 
I enabled it in gz-rendering7 in https://github.com/gazebosim/gz-rendering/pull/758